### ISSUE TITLE
High level documentation structure

### ISSUE THREAD

**Issue by c42f** [JuliaDocs/Documenter.jl#1234]
Hi everyone

I'm in the process of trying to use Documenter to document a medium sized project.

I've written plenty of technical reference material in the past (a lot in the form of function docstrings).  But that's is only one aspect of good documentation and I'm somewhat of a novice in structuring the other parts.

In that respect, I've found the following blog post *extremely* helpful in thinking about the high level structure of technical documentation in a goal-driven way:

https://www.divio.com/blog/documentation/

I like this so much because it tries to structure documentation around the goals of the reader and puts me in a more focused mindset when writing the different types of documentation.

I wondered if there would be interest in adding a link to this (and perhaps other resources) in Documenter's own documentation. And perhaps even structuring Documenter's documentation along similar lines — while it's excellent and quite complete, I sometimes find it hard to navigate.

Regardless of that I have been loving the Documenter experience so far, especially when coupled with LiveServer.jl (which is another thing I think would be nice to mention btw).

**Comment by mortenpi** [comment]
Yes, I also think this 4-level approach is a really good way to think about technical documentation, and I think it would be worth advertising in the docs (and same for LiveServer). If you have time to PR something to Documenter's docs, that would be awesome!
